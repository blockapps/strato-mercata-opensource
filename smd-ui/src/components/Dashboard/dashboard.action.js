export const PRELOAD_BLOCK_NUMBER = "PRELOAD_BLOCK_NUMBER";
export const UPDATE_BLOCK_NUMBER = "UPDATE_BLOCK_NUMBER";
export const PRELOAD_CONTRACT_COUNT = "PRELOAD_CONTRACT_COUNT";
export const UPDATE_CONTRACT_COUNT = "UPDATE_CONTRACT_COUNT";
export const PRELOAD_USERS_COUNT = "PRELOAD_USERS_COUNT";
export const UPDATE_USERS_COUNT = "UPDATE_USERS_COUNT";
export const PRELOAD_TRANSACTION_COUNT = "PRELOAD_TRANSACTION_COUNT";
export const UPDATE_TRANSACTION_COUNT = "UPDATE_TRANSACTION_COUNT";
export const PRELOAD_BLOCK_PROPAGATION = "PRELOAD_BLOCK_PROPAGATION";
export const UPDATE_BLOCK_PROPAGATION = "UPDATE_BLOCK_PROPAGATION";
export const PRELOAD_BLOCK_DIFFICULTY = "PRELOAD_BLOCK_DIFFICULTY";
export const UPDATE_BLOCK_DIFFICULTY = "UPDATE_BLOCK_DIFFICULTY";
export const PRELOAD_TRANSACTION_TYPES = "PRELOAD_TRANSACTION_TYPES";
export const UPDATE_TRANSACTION_TYPES = "UPDATE_TRANSACTION_TYPES";
export const PRELOAD_HEALTH = "PRELOAD_HEALTH";
export const UPDATE_HEALTH = "UPDATE_HEALTH";
export const CHANGE_HEALTH_STATUS = "CHANGE_HEALTH_STATUS";
export const PRELOAD_NODE_UPTIME = "PRELOAD_NODE_UPTIME";
export const UPDATE_NODE_UPTIME = "UPDATE_NODE_UPTIME";
export const PRELOAD_SYSTEM_INFO = "PRELOAD_SYSTEM_INFO";
export const UPDATE_SYSTEM_INFO = "UPDATE_SYSTEM_INFO";
export const PRELOAD_SHARD_COUNT = "PRELOAD_SHARD_COUNT";
export const UPDATE_SHARD_COUNT = "UPDATE_SHARD_COUNT";
export const PRELOAD_NETWORK_HEALTH = "PRELOAD_NETWORK_HEALTH";
export const UPDATE_NETWORK_HEALTH = "UPDATE_NETWORK_HEALTH";

// will trigger from socket.saga and update the dashboard store
export const preloadBlockNumber = function (data) {
  return {
    type: PRELOAD_BLOCK_NUMBER,
    data,
  };
};

// will trigger from socket.saga and update the dashboard store
export const updateBlockNumber = function (data) {
  return {
    type: UPDATE_BLOCK_NUMBER,
    data,
  };
};

// will trigger from socket.saga and update the dashboard store
export const preloadContractCount = function (data) {
  return {
    type: PRELOAD_CONTRACT_COUNT,
    data,
  };
};

// will trigger from socket.saga and update the dashboard store
export const updateContractCount = function (data) {
  return {
    type: UPDATE_CONTRACT_COUNT,
    data,
  };
};

// will trigger from socket.saga and update the dashboard store
export const preloadUsersCount = function (data) {
  return {
    type: PRELOAD_USERS_COUNT,
    data,
  };
};

// will trigger from socket.saga and update the dashboard store
export const updateUsersCount = function (data) {
  return {
    type: UPDATE_USERS_COUNT,
    data,
  };
};

// will trigger from socket.saga and update the dashboard store
export const preloadTransactionsCount = function (data) {
  return {
    type: PRELOAD_TRANSACTION_COUNT,
    data,
  };
};

export const updateTransactionCount = function (data) {
  return {
    type: UPDATE_TRANSACTION_COUNT,
    data,
  };
};

// will trigger from socket.saga and update the dashboard store
export const preloadBlockPropagation = function (data) {
  return {
    type: PRELOAD_BLOCK_PROPAGATION,
    data,
  };
};

export const updateBlockPropagation = function (data) {
  return {
    type: UPDATE_BLOCK_PROPAGATION,
    data,
  };
};

// will trigger from socket.saga and update the dashboard store
export const preloadBlockDifficulty = function (data) {
  return {
    type: PRELOAD_BLOCK_DIFFICULTY,
    data,
  };
};

export const updateBlockDifficulty = function (data) {
  return {
    type: UPDATE_BLOCK_DIFFICULTY,
    data,
  };
};

// will trigger from socket.saga and update the dashboard store
export const preloadTransactionType = function (data) {
  return {
    type: PRELOAD_TRANSACTION_TYPES,
    data,
  };
};

export const updateTransactionType = function (data) {
  return {
    type: UPDATE_TRANSACTION_TYPES,
    data,
  };
};

export const preloadHealth = function (data) {
  return {
    type: PRELOAD_HEALTH,
    data,
  };
};

export const updateHealth = function (data) {
  return {
    type: UPDATE_HEALTH,
    data,
  };
};

export const preloadNodeUptime = function (data) {
  return {
    type: PRELOAD_NODE_UPTIME,
    data,
  };
};

export const updateNodeUptime = function (data) {
  return {
    type: UPDATE_NODE_UPTIME,
    data,
  };
};

export const preloadSystemInfo = function (data) {
  return {
    type: PRELOAD_SYSTEM_INFO,
    data,
  };
};

export const updateSystemInfo = function (data) {
  return {
    type: UPDATE_SYSTEM_INFO,
    data,
  };
};

export const changeHealthStatus = function (data) {
  return {
    type: CHANGE_HEALTH_STATUS,
    data,
  };
};

export const preloadShardCount = (data) => {
  return {
    type: PRELOAD_SHARD_COUNT,
    data,
  };
};
export const updateShardCount = (data) => {
  return {
    type: UPDATE_SHARD_COUNT,
    data,
  };
};

export const preloadNetworkHealth = (data) => {
  return {
    type: PRELOAD_NETWORK_HEALTH,
    data,
  };
};
export const updateNetworkHealth = (data) => {
  return {
    type: UPDATE_NETWORK_HEALTH,
    data,
  };
};
