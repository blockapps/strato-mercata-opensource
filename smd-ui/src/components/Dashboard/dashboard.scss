@import "../../../node_modules/@blueprintjs/core/dist/variables.scss";

.network-status {
  color: $pt-intent-success;
  margin-top: 0.7em;
  font-size: 20px;
  font-weight: bold;
  background-color: $pt-dark-code-background-color;
  padding: 5px 10px;
  text-align: center;
}
.full-width {
  display: "inline-block";
  width: 100%;
}

.bad-metric {
  color: #f74b4b;
}

.good-metric {
  color: #f5f8fa;
}

.metric {
  display: inline-block;
}

.healthy-network {
  color: #62d96b;
}

.unhealthy-network {
  color: #ffc940;
}

#networkInfo {
  display: flex;
  flex-direction: column;
}
