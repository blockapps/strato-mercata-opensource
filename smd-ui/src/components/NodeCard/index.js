import React, { Component } from 'react';
import { connect } from 'react-redux';
import { withRouter } from 'react-router-dom';
import { Collapse } from '@blueprintjs/core';
import './nodeCard.css';
import PeersCard from '../PeersCard';
import HexText from '../HexText';
import { subscribeRoom, unSubscribeRoom } from '../../sockets/socket.actions'
import {
  GET_PEERS
} from '../../sockets/rooms'

class NodeCard extends Component {

  constructor() {
    super()
    this.state = {
      isOpen: false
    }
  }

  componentDidMount() {
    this.props.subscribeRoom(GET_PEERS)
  }

  componentWillUnmount() {
    this.props.unSubscribeRoom(GET_PEERS)
  }

  handleClick = () => {
    this.setState({ isOpen: !this.state.isOpen });
  };

  render() {
    const node = this.props.node;
    const peers = node.peers ? Object.getOwnPropertyNames(node.peers) : [];
    const blockNumber = this.props.dashboard.lastBlockNumber;
    let className = 'pt-card pt-elevation-2 node-success pt-interactive';
    let arrowIcon = 'col-xs-3 text-right pt-icon-standard '
    arrowIcon += this.state.isOpen ? 'pt-icon-caret-up' : 'pt-icon-caret-down'

    return (
      <div>
        <div className={className} onClick={this.handleClick}>
          <div className="row">
            <div className='col-sm-6'>

              <h3>Peers ({peers.length})</h3>
              <span className={arrowIcon}>{this.state.isOpen ? "Close" : "Expand"}</span>
            </div>
          </div>
        <hr/>
        <Collapse isOpen={this.state.isOpen}>
          <PeersCard />
        </Collapse>
        </div>
      </div>
    );
  }
}

export function mapStateToProps(state) {
  return {
    dashboard: state.dashboard,
    node: state.node
  };
}

export default withRouter(
  connect(
    mapStateToProps,
    {
      subscribeRoom,
      unSubscribeRoom,
    }
  )(NodeCard)
);