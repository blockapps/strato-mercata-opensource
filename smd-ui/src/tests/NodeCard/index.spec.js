import React from 'react';
import NodeCard, { mapStateToProps } from "../../components/NodeCard/index";
import { dashboard, node, initialState } from '../Dashboard/dashboardMock';

describe('NodeCard: index', () => {

  describe('render component', () => {

    test('without values', () => {
      const props = {
        node: {
          "name": "LOCALHOST",
          "peers": null,
          "coinbase": null
        },
        dashboard: initialState,
        subscribeRoom: jest.fn(),
        unSubscribeRoom: jest.fn()
      };
      const wrapper = shallow(
        <NodeCard.WrappedComponent {...props} />
      );

      expect(wrapper).toMatchSnapshot();
    });

    test('with values', () => {
      const props = {
        dashboard,
        node,
        subscribeRoom: jest.fn(),
        unSubscribeRoom: jest.fn()
      };

      const wrapper = shallow(
        <NodeCard.WrappedComponent {...props} />
      );

      expect(wrapper).toMatchSnapshot();
    });

  });

  test('expand card on click', () => {
    const props = {
      dashboard,
      node,
      subscribeRoom: jest.fn(),
      unSubscribeRoom: jest.fn()
    };

    const wrapper = shallow(
      <NodeCard.WrappedComponent {...props} />
    );

    expect(wrapper.state('isOpen')).toBe(false);
    wrapper.find('div').at(1).simulate('click');
    expect(wrapper.state('isOpen')).toBe(true);
    wrapper.find('div').at(1).simulate('click');
    expect(wrapper.state('isOpen')).toBe(false);
  });

  test('valid functions as a props', () => {
    const props = {
      dashboard,
      node,
      subscribeRoom: jest.fn().mockReturnValue('subscribeRoom'),
      unSubscribeRoom: jest.fn().mockReturnValue('unSubscribeRoom')
    };

    const wrapper = shallow(
      <NodeCard.WrappedComponent {...props} />
    );

    expect(wrapper.instance().props.subscribeRoom()).toBe('subscribeRoom');
    expect(wrapper.instance().props.unSubscribeRoom()).toBe('unSubscribeRoom');
  });

  test('componentDidMount', () => {
    const props = {
      dashboard,
      node,
      subscribeRoom: jest.fn(),
      unSubscribeRoom: jest.fn()
    };

    const wrapper = shallow(
      <NodeCard.WrappedComponent {...props} />
    );

    expect(props.subscribeRoom).toHaveBeenCalled();
    expect(props.subscribeRoom.mock.calls).toEqual([["GET_PEERS"]]);
  });

  test('componentWillUnmount', () => {
    const props = {
      dashboard,
      node,
      subscribeRoom: jest.fn(),
      unSubscribeRoom: jest.fn(),
    };

    const wrapper = shallow(
      <NodeCard.WrappedComponent {...props} />
    );

    wrapper.unmount();
    expect(props.unSubscribeRoom).toHaveBeenCalled();
    expect(props.unSubscribeRoom.mock.calls).toEqual([["GET_PEERS"]]);
  });

  test('mapStateToProps', () => {
    const state = {
      dashboard,
      node
    };

    expect(mapStateToProps(state)).toMatchSnapshot();
  });

});