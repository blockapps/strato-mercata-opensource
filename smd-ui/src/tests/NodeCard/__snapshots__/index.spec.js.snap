// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NodeCard: index mapStateToProps 1`] = `
Object {
  "dashboard": Object {
    "blockDifficulty": Array [
      Object {
        "x": 0,
        "y": 8192,
      },
      Object {
        "x": 1,
        "y": 131072,
      },
      Object {
        "x": 2,
        "y": 131072,
      },
      Object {
        "x": 3,
        "y": 131072,
      },
      Object {
        "x": 4,
        "y": 131072,
      },
      Object {
        "x": 5,
        "y": 131072,
      },
      Object {
        "x": 6,
        "y": 131072,
      },
      Object {
        "x": 7,
        "y": 131072,
      },
      Object {
        "x": 8,
        "y": 131072,
      },
      Object {
        "x": 9,
        "y": 131072,
      },
    ],
    "blockPropagation": Array [
      Object {
        "x": 0,
        "y": 0,
      },
      Object {
        "x": 1,
        "y": 1512994588,
      },
      Object {
        "x": 2,
        "y": 3358,
      },
      Object {
        "x": 3,
        "y": 19,
      },
      Object {
        "x": 4,
        "y": 1846,
      },
      Object {
        "x": 5,
        "y": 301,
      },
      Object {
        "x": 6,
        "y": 48,
      },
      Object {
        "x": 7,
        "y": 55429,
      },
      Object {
        "x": 8,
        "y": 106195,
      },
      Object {
        "x": 9,
        "y": 58,
      },
    ],
    "contractsCount": 2,
    "lastBlockNumber": 9,
    "transactionTypes": Array [
      Object {
        "type": "FunctionCall",
        "val": 6,
      },
      Object {
        "type": "Transfer",
        "val": 1,
      },
      Object {
        "type": "Contract",
        "val": 2,
      },
    ],
    "transactionsCount": Array [
      Object {
        "x": 0,
        "y": 0,
      },
      Object {
        "x": 1,
        "y": 1,
      },
      Object {
        "x": 2,
        "y": 1,
      },
      Object {
        "x": 3,
        "y": 1,
      },
      Object {
        "x": 4,
        "y": 1,
      },
      Object {
        "x": 5,
        "y": 1,
      },
      Object {
        "x": 6,
        "y": 1,
      },
      Object {
        "x": 7,
        "y": 1,
      },
      Object {
        "x": 8,
        "y": 1,
      },
      Object {
        "x": 9,
        "y": 1,
      },
    ],
    "usersCount": 1,
  },
  "node": Object {
    "coinbase": Object {
      "coinbase": "82069cc441c42b2706fa5d3129deeb677812e5a5",
    },
    "name": "LOCALHOST",
    "peers": Object {},
  },
}
`;

exports[`NodeCard: index render component with values 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <NodeCard
    dashboard={
      Object {
        "blockDifficulty": Array [
          Object {
            "x": 0,
            "y": 8192,
          },
          Object {
            "x": 1,
            "y": 131072,
          },
          Object {
            "x": 2,
            "y": 131072,
          },
          Object {
            "x": 3,
            "y": 131072,
          },
          Object {
            "x": 4,
            "y": 131072,
          },
          Object {
            "x": 5,
            "y": 131072,
          },
          Object {
            "x": 6,
            "y": 131072,
          },
          Object {
            "x": 7,
            "y": 131072,
          },
          Object {
            "x": 8,
            "y": 131072,
          },
          Object {
            "x": 9,
            "y": 131072,
          },
        ],
        "blockPropagation": Array [
          Object {
            "x": 0,
            "y": 0,
          },
          Object {
            "x": 1,
            "y": 1512994588,
          },
          Object {
            "x": 2,
            "y": 3358,
          },
          Object {
            "x": 3,
            "y": 19,
          },
          Object {
            "x": 4,
            "y": 1846,
          },
          Object {
            "x": 5,
            "y": 301,
          },
          Object {
            "x": 6,
            "y": 48,
          },
          Object {
            "x": 7,
            "y": 55429,
          },
          Object {
            "x": 8,
            "y": 106195,
          },
          Object {
            "x": 9,
            "y": 58,
          },
        ],
        "contractsCount": 2,
        "lastBlockNumber": 9,
        "transactionTypes": Array [
          Object {
            "type": "FunctionCall",
            "val": 6,
          },
          Object {
            "type": "Transfer",
            "val": 1,
          },
          Object {
            "type": "Contract",
            "val": 2,
          },
        ],
        "transactionsCount": Array [
          Object {
            "x": 0,
            "y": 0,
          },
          Object {
            "x": 1,
            "y": 1,
          },
          Object {
            "x": 2,
            "y": 1,
          },
          Object {
            "x": 3,
            "y": 1,
          },
          Object {
            "x": 4,
            "y": 1,
          },
          Object {
            "x": 5,
            "y": 1,
          },
          Object {
            "x": 6,
            "y": 1,
          },
          Object {
            "x": 7,
            "y": 1,
          },
          Object {
            "x": 8,
            "y": 1,
          },
          Object {
            "x": 9,
            "y": 1,
          },
        ],
        "usersCount": 1,
      }
    }
    node={
      Object {
        "coinbase": Object {
          "coinbase": "82069cc441c42b2706fa5d3129deeb677812e5a5",
        },
        "name": "LOCALHOST",
        "peers": Object {},
      }
    }
    subscribeRoom={[Function]}
    unSubscribeRoom={[Function]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <div
        className="pt-card pt-elevation-2 node-success pt-interactive"
        onClick={[Function]}
      >
        <div
          className="row"
        >
          <div
            className="col-sm-6"
          >
            <h3>
              Peers (
              0
              )
            </h3>
            <span
              className="col-xs-3 text-right pt-icon-standard pt-icon-caret-down"
            >
              Expand
            </span>
          </div>
        </div>
        <hr />
        <Blueprint.Collapse
          component="div"
          isOpen={false}
          keepChildrenMounted={false}
          transitionDuration={200}
        >
          <withRouter(Connect(PeersCard)) />
        </Blueprint.Collapse>
      </div>,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="row"
          >
            <div
              className="col-sm-6"
            >
              <h3>
                Peers (
                0
                )
              </h3>
              <span
                className="col-xs-3 text-right pt-icon-standard pt-icon-caret-down"
              >
                Expand
              </span>
            </div>
          </div>,
          <hr />,
          <Blueprint.Collapse
            component="div"
            isOpen={false}
            keepChildrenMounted={false}
            transitionDuration={200}
          >
            <withRouter(Connect(PeersCard)) />
          </Blueprint.Collapse>,
        ],
        "className": "pt-card pt-elevation-2 node-success pt-interactive",
        "onClick": [Function],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <div
              className="col-sm-6"
            >
              <h3>
                Peers (
                0
                )
              </h3>
              <span
                className="col-xs-3 text-right pt-icon-standard pt-icon-caret-down"
              >
                Expand
              </span>
            </div>,
            "className": "row",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <h3>
                  Peers (
                  0
                  )
                </h3>,
                <span
                  className="col-xs-3 text-right pt-icon-standard pt-icon-caret-down"
                >
                  Expand
                </span>,
              ],
              "className": "col-sm-6",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    "Peers (",
                    0,
                    ")",
                  ],
                },
                "ref": null,
                "rendered": Array [
                  "Peers (",
                  0,
                  ")",
                ],
                "type": "h3",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Expand",
                  "className": "col-xs-3 text-right pt-icon-standard pt-icon-caret-down",
                },
                "ref": null,
                "rendered": "Expand",
                "type": "span",
              },
            ],
            "type": "div",
          },
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": "hr",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": <withRouter(Connect(PeersCard)) />,
            "component": "div",
            "isOpen": false,
            "keepChildrenMounted": false,
            "transitionDuration": 200,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {},
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": [Function],
        },
      ],
      "type": "div",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <div
          className="pt-card pt-elevation-2 node-success pt-interactive"
          onClick={[Function]}
        >
          <div
            className="row"
          >
            <div
              className="col-sm-6"
            >
              <h3>
                Peers (
                0
                )
              </h3>
              <span
                className="col-xs-3 text-right pt-icon-standard pt-icon-caret-down"
              >
                Expand
              </span>
            </div>
          </div>
          <hr />
          <Blueprint.Collapse
            component="div"
            isOpen={false}
            keepChildrenMounted={false}
            transitionDuration={200}
          >
            <withRouter(Connect(PeersCard)) />
          </Blueprint.Collapse>
        </div>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <div
              className="row"
            >
              <div
                className="col-sm-6"
              >
                <h3>
                  Peers (
                  0
                  )
                </h3>
                <span
                  className="col-xs-3 text-right pt-icon-standard pt-icon-caret-down"
                >
                  Expand
                </span>
              </div>
            </div>,
            <hr />,
            <Blueprint.Collapse
              component="div"
              isOpen={false}
              keepChildrenMounted={false}
              transitionDuration={200}
            >
              <withRouter(Connect(PeersCard)) />
            </Blueprint.Collapse>,
          ],
          "className": "pt-card pt-elevation-2 node-success pt-interactive",
          "onClick": [Function],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <div
                className="col-sm-6"
              >
                <h3>
                  Peers (
                  0
                  )
                </h3>
                <span
                  className="col-xs-3 text-right pt-icon-standard pt-icon-caret-down"
                >
                  Expand
                </span>
              </div>,
              "className": "row",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <h3>
                    Peers (
                    0
                    )
                  </h3>,
                  <span
                    className="col-xs-3 text-right pt-icon-standard pt-icon-caret-down"
                  >
                    Expand
                  </span>,
                ],
                "className": "col-sm-6",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": Array [
                      "Peers (",
                      0,
                      ")",
                    ],
                  },
                  "ref": null,
                  "rendered": Array [
                    "Peers (",
                    0,
                    ")",
                  ],
                  "type": "h3",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Expand",
                    "className": "col-xs-3 text-right pt-icon-standard pt-icon-caret-down",
                  },
                  "ref": null,
                  "rendered": "Expand",
                  "type": "span",
                },
              ],
              "type": "div",
            },
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {},
            "ref": null,
            "rendered": null,
            "type": "hr",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": <withRouter(Connect(PeersCard)) />,
              "component": "div",
              "isOpen": false,
              "keepChildrenMounted": false,
              "transitionDuration": 200,
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {},
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            "type": [Function],
          },
        ],
        "type": "div",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactFifteenAdapter {
      "options": Object {
        "supportPrevContextArgumentOfComponentDidUpdate": true,
      },
    },
  },
}
`;

exports[`NodeCard: index render component without values 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <NodeCard
    dashboard={
      Object {
        "blockDifficulty": Array [],
        "blockPropagation": Array [],
        "contractsCount": 0,
        "healthStatus": false,
        "lastBlockNumber": 0,
        "shardCount": 0,
        "transactionTypes": Array [],
        "transactionsCount": Array [],
        "uptime": 0,
        "usersCount": 0,
      }
    }
    node={
      Object {
        "coinbase": null,
        "name": "LOCALHOST",
        "peers": null,
      }
    }
    subscribeRoom={[Function]}
    unSubscribeRoom={[Function]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <div
        className="pt-card pt-elevation-2 node-success pt-interactive"
        onClick={[Function]}
      >
        <div
          className="row"
        >
          <div
            className="col-sm-6"
          >
            <h3>
              Peers (
              0
              )
            </h3>
            <span
              className="col-xs-3 text-right pt-icon-standard pt-icon-caret-down"
            >
              Expand
            </span>
          </div>
        </div>
        <hr />
        <Blueprint.Collapse
          component="div"
          isOpen={false}
          keepChildrenMounted={false}
          transitionDuration={200}
        >
          <withRouter(Connect(PeersCard)) />
        </Blueprint.Collapse>
      </div>,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="row"
          >
            <div
              className="col-sm-6"
            >
              <h3>
                Peers (
                0
                )
              </h3>
              <span
                className="col-xs-3 text-right pt-icon-standard pt-icon-caret-down"
              >
                Expand
              </span>
            </div>
          </div>,
          <hr />,
          <Blueprint.Collapse
            component="div"
            isOpen={false}
            keepChildrenMounted={false}
            transitionDuration={200}
          >
            <withRouter(Connect(PeersCard)) />
          </Blueprint.Collapse>,
        ],
        "className": "pt-card pt-elevation-2 node-success pt-interactive",
        "onClick": [Function],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <div
              className="col-sm-6"
            >
              <h3>
                Peers (
                0
                )
              </h3>
              <span
                className="col-xs-3 text-right pt-icon-standard pt-icon-caret-down"
              >
                Expand
              </span>
            </div>,
            "className": "row",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <h3>
                  Peers (
                  0
                  )
                </h3>,
                <span
                  className="col-xs-3 text-right pt-icon-standard pt-icon-caret-down"
                >
                  Expand
                </span>,
              ],
              "className": "col-sm-6",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    "Peers (",
                    0,
                    ")",
                  ],
                },
                "ref": null,
                "rendered": Array [
                  "Peers (",
                  0,
                  ")",
                ],
                "type": "h3",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Expand",
                  "className": "col-xs-3 text-right pt-icon-standard pt-icon-caret-down",
                },
                "ref": null,
                "rendered": "Expand",
                "type": "span",
              },
            ],
            "type": "div",
          },
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": "hr",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": <withRouter(Connect(PeersCard)) />,
            "component": "div",
            "isOpen": false,
            "keepChildrenMounted": false,
            "transitionDuration": 200,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {},
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": [Function],
        },
      ],
      "type": "div",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <div
          className="pt-card pt-elevation-2 node-success pt-interactive"
          onClick={[Function]}
        >
          <div
            className="row"
          >
            <div
              className="col-sm-6"
            >
              <h3>
                Peers (
                0
                )
              </h3>
              <span
                className="col-xs-3 text-right pt-icon-standard pt-icon-caret-down"
              >
                Expand
              </span>
            </div>
          </div>
          <hr />
          <Blueprint.Collapse
            component="div"
            isOpen={false}
            keepChildrenMounted={false}
            transitionDuration={200}
          >
            <withRouter(Connect(PeersCard)) />
          </Blueprint.Collapse>
        </div>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <div
              className="row"
            >
              <div
                className="col-sm-6"
              >
                <h3>
                  Peers (
                  0
                  )
                </h3>
                <span
                  className="col-xs-3 text-right pt-icon-standard pt-icon-caret-down"
                >
                  Expand
                </span>
              </div>
            </div>,
            <hr />,
            <Blueprint.Collapse
              component="div"
              isOpen={false}
              keepChildrenMounted={false}
              transitionDuration={200}
            >
              <withRouter(Connect(PeersCard)) />
            </Blueprint.Collapse>,
          ],
          "className": "pt-card pt-elevation-2 node-success pt-interactive",
          "onClick": [Function],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <div
                className="col-sm-6"
              >
                <h3>
                  Peers (
                  0
                  )
                </h3>
                <span
                  className="col-xs-3 text-right pt-icon-standard pt-icon-caret-down"
                >
                  Expand
                </span>
              </div>,
              "className": "row",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <h3>
                    Peers (
                    0
                    )
                  </h3>,
                  <span
                    className="col-xs-3 text-right pt-icon-standard pt-icon-caret-down"
                  >
                    Expand
                  </span>,
                ],
                "className": "col-sm-6",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": Array [
                      "Peers (",
                      0,
                      ")",
                    ],
                  },
                  "ref": null,
                  "rendered": Array [
                    "Peers (",
                    0,
                    ")",
                  ],
                  "type": "h3",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Expand",
                    "className": "col-xs-3 text-right pt-icon-standard pt-icon-caret-down",
                  },
                  "ref": null,
                  "rendered": "Expand",
                  "type": "span",
                },
              ],
              "type": "div",
            },
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {},
            "ref": null,
            "rendered": null,
            "type": "hr",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": <withRouter(Connect(PeersCard)) />,
              "component": "div",
              "isOpen": false,
              "keepChildrenMounted": false,
              "transitionDuration": 200,
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {},
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            "type": [Function],
          },
        ],
        "type": "div",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactFifteenAdapter {
      "options": Object {
        "supportPrevContextArgumentOfComponentDidUpdate": true,
      },
    },
  },
}
`;
